packages/grafana-ui/src/components/Button/Button.tsx

```diff
@@ -9,11 +9,14 @@ import { getFocusStyles, getMouseFocusStyles } from '../../themes/mixins';
 import { Icon } from '../Icon/Icon';
 
 export type ButtonVariant = 'primary' | 'secondary' | 'destructive' | 'link';
-export const allButtonVariants: ButtonVariant[] = ['primary', 'secondary', 'destructive', 'link'];
+export const allButtonVariants: ButtonVariant[] = ['primary', 'secondary', 'destructive'];
+export type ButtonStyle = 'solid' | 'outline' | 'text';
+export const allButtonStyles: ButtonStyle[] = ['solid', 'outline', 'text'];
```

Comment:
fwiw if you wanted to make it a bit simpler and less error prone you could use derived types:

```typescript
export const allButtonVariants = ['primary', 'secondary', 'destructive'] as const;
export type ButtonVariant = typeof allButtonVariants[number]

export const allButtonStyles = ['solid', 'outline', 'text'] as const;
export type ButtonStyle = typeof allButtonStyles[number]

```